Sources/Direction.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  Direction.swift
+//  Euclid
+//
+//  Created by Ioannis Kaliakatsos on 02.12.2020.
+//  Copyright © 2020 Nick Lockwood. All rights reserved.
+//
+
+import Foundation
+
+public struct Direction: AdditiveArithmeticCartesianComponentsRepresentable {
+    public let x: Double
+    public let y: Double
+    public let z: Double
+
+    public init(x: Double = 0, y: Double = 0, z: Double = 0) {
+        let componentsNorm = (x * x + y * y + z * z).squareRoot()
+        if componentsNorm != 0 {
+            self.x = x / componentsNorm
+            self.y = y / componentsNorm
+            self.z = z / componentsNorm
+        } else {
+            self.x = 0
+            self.y = 0
+            self.z = 0
+        }
+    }
+}
+
+public extension Direction {
+    static let zero = Direction()
```

Comment:
When would you use a zero direction?

Reply:
As a fallback in the case where the moving direction is complicated and needs to be calculated in a separate function, however no meaningful direction can be found. Perhaps you are right, in this case we can return ```nil``` from this hypothetical function.

Reply:
Now I remembered why: `Distance` has a member called `direction`, which returns the normalized direction for a given distance vector. Say you have two identical points `p1` and `p2` of type `Position`, so you construct the distance vector `p12 = p2 - p1`. 

You want to know the direction of this vector, so calling the `.direction` should return the `zero` direction. This would be also valid mathematically, this is the zero vector; it is parallel but at the same time normal to any other vector.

This fact is heavily used in my branch [line-plane](https://github.com/jkalias/Euclid/tree/line-plane), which is based on this one. There you can see for example how `Position`, `Distance` and `Direction` are used in general 3D geometric calculations.

Let me know what you think of this `zero` direction